src/processing/script/index.js

```diff
@@ -120,6 +120,8 @@ export function applyChanges (script, changes, isObject) {
     if (!changes.length)
         return script;
 
+    changes.sort((a, b) => a.start - b.start);
```

Comment:
Well, it's not. I can't figure it out still, just your verbal description helped somehow. Also, when I see code I don't see the test. And BTW, have you tried string concat instead of arr.join for chunks? In V8 concatenation is optimized and should be more robust.